clippy_utils/src/check_proc_macro.rs

```diff
@@ -250,6 +250,27 @@ fn variant_search_pat(v: &Variant<'_>) -> (Pat, Pat) {
     }
 }
 
+fn fn_kind_pat(tcx: TyCtxt<'_>, kind: &FnKind<'_>, body: &Body<'_>, hir_id: HirId) -> (Pat, Pat) {
+    let (start_pat, end_pat, visibility) = match kind {
+        FnKind::ItemFn(.., header) => (
+            fn_header_search_pat(*header),
+            Pat::Str(""),
+            tcx.visibility(tcx.hir().local_def_id(hir_id)),
```

Comment:
Previously visibility was provided in `FnKind`, this isn't the case anymore, so this is kinda workaround to get it. Not sure if this is right or efficient.

Reply:
You can get the HIR node for the item (through `cx.tcx.hir()`) and use the `vis_span` field. If the span isn't empty it will start with `pub`. I don't think `Visibility` gives you an easy way to do this.

Reply:
I'm not entirely following, do you mean `tcx.hir().expect_item(tcx.hir().local_def_id(hir_id)).vis_span`?

Reply:
Use `tcx.hir().get(hir_id)`. It should give `Node::Item`/`ImplItem`/`TraitItem`.

Reply:
Done! I hope this is correct now.